cmd/system-probe/api/module/loader.go

```diff
@@ -64,18 +70,53 @@ func Register(cfg *config.Config, httpMux *http.ServeMux, factories []Factory) e
 
 // GetStats returns the stats from all modules, namespaced by their names
 func GetStats() map[config.ModuleName]interface{} {
+	l.Lock()
+	defer l.Unlock()
+
 	stats := make(map[config.ModuleName]interface{})
 	for name, module := range l.modules {
 		stats[name] = module.GetStats()
 	}
 	return stats
 }
 
+// RestartModule triggers a module restart
+func RestartModule(factory Factory) error {
+	l.Lock()
+	defer l.Unlock()
+
+	currentModule := l.modules[factory.Name]
```

Comment:
should we not check `l.closed` here in case of pending request released by the lock just after the `Close()`

Reply:
done @safchain